codec/unknownproto/unknown_fields.go

```diff
@@ -92,6 +92,11 @@ func RejectUnknownFields(bz []byte, msg proto.Message, allowUnknownNonCriticals
 		// Skip over the bytes that store fieldNumber and wireType bytes.
 		bz = bz[m:]
 		n := protowire.ConsumeFieldValue(tagNum, wireType, bz)
+		if n < 0 {
+			err = fmt.Errorf("unknownproto: %w: could not consume field value for tagNum: %d, wireType: %q",
```

Comment:
Typically, we wrap errors at the end.

```go
  return fmt.Errorf("failed to consume field value for tagNum: %d, wireType: %q; %w", ...)
```

Also, please avoid naked returns and just `return fmt.Errorf("...")`

Reply:
>Also, please avoid naked returns and just return fmt.Errorf("...")

The return has another value, and all of them are named returns hence the pattern, but sure.

>Typically, we wrap errors at the end.

I can change that if that's the pattern in these libraries, but the wrapping was meant to provide a meaningful context of the error:
`unknown proto: unexpected EOF: could not consume....`
instead of the suggestion
`failed to consume field value for tagNum: ...; unexpected EOF`

Reply:
Wrapping always occurs with the error at the end. This is the standard of the SDK (i.e. `"<context/msg>: <error>"`) [ref](https://blog.golang.org/go1.13-errors).

> The return has another value, and all of them are named returns hence the pattern, but sure.

That's fine. Just return all values. Naked returns must be avoided at all costs unless the function or method is very trivial and/or concise (which this one certainly isn't 😃 ).